src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc

```diff
@@ -223,6 +223,15 @@ void GetInteriorVehicleDataRequest::Execute() {
     return;
   }
   ProcessResponseToMobileFromCache(app);
+  if (AppShouldBeUnsubscribed()) {
```

Comment:
we are talking about the subscription, but here I can see Unsubscribed ?

Reply:
I believe it should be smthg like 
```
  if (TheLastAppShouldBeUnsubscribed(app) ||
      !interior_data_cache_.Contains(module)) || FirstAppWantsToSubscribe(.. ) {
     ... 
     SendHMIRequest(hmi_apis::FunctionID::RC_GetInteriorVehicleData, ...
     }
```

Reply:
I don't understand this logic while reading the code. 

Looks something like : 

So we send the subscription request to HMI if :
1. data is not cached 
2. App want's to subscribe, but no other app subscribed

So we send the unsubscription request to HMI if :
1. data is not cached 
2. App want's to unsubscribe and this is the last subscribed app.


Please split the flow to subscription and unsubscription with appropriate function names.